### ISSUE TITLE
Reproduction of paper results

### ISSUE THREAD

**Issue by Vazzz007** [MCG-NJU/MOC-Detector#15]
Hi,
First of all, thanks for open source code!

I have an issue with reproduction of your results on UCF dataset. I got the following results after training and inference:
RGB only (scores are located as follows: FrameAP@0.5 | VideoAP@0.2 | @0.5 | @0.75 |  0.5:0.95):
(paper) ucf_dla34_K7_rgb_coco.pth: **73.14** | 78.81 |   51.02 | 27.05 | 26.51

ucf_dla34_K7_rgb_coco with N10, w/o flip_test:  69.84 | 75.77 | 47.55 | 24.57 | 24.76

ucf_dla34_K7_rgb_coco with N100 and flip_test: 72.13 | 78.25 | 50.48 | 26.18 | 26.32

RGB + FLOW:
(paper) K7   RGB + FLOW COCO : **78.01** | 82.81 | 53.83 | 29.59 | 28.33

K7 RGB + FLOW COCO with N10, w/o flip_test:  74.98 | 80.88 | 52.82 | 27.39 | 27.30

K7 RGB + FLOW COCO with N100 and flip_test: 76.52 | 81.62 | 53.52 | 29.05 | 28.05

So, as you can see my results are far enough from yours. I mean you wrote that while inference N10 w/o flip_test is faster (no doubt) and scores are pretty much the same than N100 with flip_test. But I get the opposite.

It would be nice if you help me to understand what am I doing wrong.

Some info that might be useful:
For training and evaluation I've used master branch code.
All parameters are the same as in the script MOC-Detector/scripts/train_ucf_k7_dla.sh
I use PyTorch 1.3 and python 3.6
I've tried to train with enabled cudnn lib for batchnorm - the same problem.

**Comment by ArchiZX** [comment]
It seems that your reproduction results are lower than ours for `K7 RGB`. I guess the FLOW results may have the same problem.

Our experimental settings are `Python 3.5.2, PyTorch 0.4.1 with cuda9.0` for `K7 RGB+FLOW`. For pytorch1.6 we only tested the inference results without reproducing it.

Would you mind using pytorch0.4.1 to reproduce it again?

BTW, the performance gap is about 0.5-2.0 for N10 w/o flip for our released model (depends on the evaluation metric).

**Comment by Vazzz007** [comment]
Thanks for fast response. Ok, I'll try with PyTorch 0.4.1

**Comment by Vazzz007** [comment]
Hi!

I've tried with recommended parameters to train rgb model. Results are the same:
(scores are located as follows: FrameAP@0.5 | VideoAP@0.2 | @0.5 | @0.75 | 0.5:0.95):
(paper) ucf_dla34_K7_rgb_coco.pth: 73.14 | 78.81 | 51.02 | 27.05 | 26.51

ucf_dla34_K7_rgb_coco_N100_flip 71.56 | 78.40 | 50.24 | 26.18 | 26.16

ucf_dla34_K7_rgb_coco_N10 70.24 | 77.78 | 47.64 | 25.22 | 24.61

Again, I'm more curious about why inference with N100 + flip significantly better than N10 w/o flip which is opposite to your results. And variance of scores is also a bit high. I checked this on different machines. So I don't know where to look.
Hope you give me some advice.
Thank you again

**Comment by ArchiZX** [comment]
For our RGB model, the experimental results are:
ucf_dla34_K7_rgb_coco_N100_flip: 73.14 | 78.81 | 51.02 | 27.05 | 26.51
ucf_dla34_K7_rgb_coco_N10:          72.05 | 78.23 | 50.77 | 26.10 | 26.16
(The performance gap is 0.5-1.1)

It seems that your reproduction results are a little lower than ours.(For example, the frameAP@0.5 for RGB K=7 should be at least 72.xx. And I got the similar results on my machine.)

As for `N10 w/o flip`, I wonder this may be just a coincidence for my own model.
I have not mentioned it in our paper because it was just an accidental discovery when I organized the code after the paper is accepted.



Here I give you more experimental results:
`K=5 RGB`
ucf_dla34_K5_rgb_coco_N100_flip: 71.63 | 77.74 | 49.55 | 27.04 | 26.09
ucf_dla34_K5_rgb_coco_N10:          70.10 | 76.08 | 49.14 | 26.28 | 25.70

`ImageNet pretrain`
ucf_dla34_K7_rgb_flow_ImageNet_N100_flip: 76.92 | 81.26 | 54.43 | 29.49 | 28.42
ucf_dla34_K7_rgb_flow_ImageNet_N10:          76.03 | 80.75 | 53.47 | 27.16 | 27.52

I this this performance gap is acceptable because it significantly boosts the inference speed.


Thanks for your correction and I will revise it.

**Comment by Vazzz007** [comment]
Thank you so much. It helps me.
